projects/robots/husarion/rosbot_xl/protos/RosbotXl.proto

```diff
@@ -0,0 +1,254 @@
+#VRML_SIM R2023b utf8
+# license: Apache License 2.0
+# license url: http://www.apache.org/licenses/LICENSE-2.0
+# This is a proto file for Webots for the RosbotXl
+
+EXTERNPROTO "webots://projects/devices/slamtec/protos/RpLidarA2.proto"
+EXTERNPROTO "webots://projects/devices/tdk/protos/Mpu-9250.proto"
+
+PROTO RosbotXl [
+  field  SFVec3f     translation     0 0 0
+  field  SFRotation  rotation        0 0 1 0
+  field  SFString    name            "rosbot_xl" # Is `Robot.name`.
+  field  SFString    controller      "rosbot_xl" # Is `Robot.controller`.
+  field  MFString    controllerArgs  []          # Is `Robot.controllerArgs`.
+  field  SFString    customData      ""          # Is `Robot.customData`.
+  field  SFBool      supervisor      FALSE       # Is `Robot.supervisor`.
+  field  SFBool      synchronization TRUE        # Is `Robot.synchronization`.
+  field  SFBool      selfCollision   FALSE       # Is `Robot.selfCollision`.
+  field  MFNode      lidarSlot       [ RpLidarA2 { name "laser" } ] # Extends the robot with a RpLidar A2 lidar.
+]
+{
+  Robot {
+    translation IS translation
+    rotation IS rotation
+    name IS name
+    controller IS controller
+    controllerArgs IS controllerArgs
+    customData IS customData
+    supervisor IS supervisor
+    synchronization IS synchronization
+    selfCollision IS selfCollision
+    physics Physics {
+      density -1
+      mass 3.500000
+      centerOfMass [ -0.016319 -0.004036 0.028190 ]
+      inertiaMatrix [
+        2.540425e-02 2.874642e-02 4.622710e-02
+        -2.000608e-04 -9.754257e-04 -2.603326e-04
+      ]
+    }
+    boundingObject DEF body_colision Group {
+      children [
+        DEF MAIN_BODY_BOX Transform {
+          translation 0.000000 0.000000 0.048
+          rotation 0.000000 0.000000 1.000000 0.000000
+          children [
+            Mesh {
+              url "meshes/body_collision.stl"
```

Comment:
In Webots it's not considered good practice to use a mesh as boundingObject, as it's heavy to compute collisions and less reliable. Ideally it would be better to approximate the shape of the body of the robot using primitive shapes as done in the rosbot proto.

Reply:
I understand this but we used the same concept in ignition-gazebo to exactly check the collision with the manipulator in MoveIt 2.
Maybe in the feature we would like to add the open manipulator on ROSbot XL in webots.

Reply:
It was just a suggestion to let you know what we consider "good practice", up to you

Reply:
It is often possible to closely approximate the actual bounding object with a composite bounding object made of a several basic primitives (boxes, spheres, cylinders, capsules) arranged together. This usually yields to a much better performance and better stability than a mesh.

Reply:
Should I try to change it or left the collision as a mesh?

Reply:
Okay so at this time I left this. I will change it when there are problems.